From 6b32f2fd61fab7e58125403b6e83a09a58a07dc6 Mon Sep 17 00:00:00 2001
From: Vincent Cheah Beng Keat <vincent.beng.keat.cheah@intel.com>
Date: Thu, 1 Aug 2024 18:21:03 +0800
Subject: [PATCH] Set Scanout by default

recommend to flag all potential scanout surface as SCNAOUT. This is
required to fix DMA DRM render on XE.

Upstream-Status: Submitted [https://github.com/intel/media-driver/pull/1822]

Signed-off-by: Cheah, Vincent Beng Keat <vincent.beng.keat.cheah@intel.com>
---
 media_softlet/linux/common/os/xe/mos_bufmgr_xe.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media_softlet/linux/common/os/xe/mos_bufmgr_xe.c b/media_softlet/linux/common/os/xe/mos_bufmgr_xe.c
index 9bc065eaa..26b635604 100644
--- a/media_softlet/linux/common/os/xe/mos_bufmgr_xe.c
+++ b/media_softlet/linux/common/os/xe/mos_bufmgr_xe.c
@@ -1321,6 +1321,10 @@ mos_bo_alloc_xe(struct mos_bufmgr *bufmgr,
      */
     create.cpu_caching = alloc->ext.cpu_cacheable ? DRM_XE_GEM_CPU_CACHING_WB : DRM_XE_GEM_CPU_CACHING_WC;
 
+    if ((strcmp(alloc->name, "MEDIA") == 0 || strcmp(alloc->name, "Media") == 0)
+        && create.cpu_caching == DRM_XE_GEM_CPU_CACHING_WC)
+            create.flags |= DRM_XE_GEM_CREATE_FLAG_SCANOUT;
+
     ret = drmIoctl(bufmgr_gem->fd,
         DRM_IOCTL_XE_GEM_CREATE,
         &create);
-- 
2.34.1

